src/main/java/com/google/cloud/genomics/dataflow/utils/GCSHelper.java

```diff
@@ -0,0 +1,202 @@
+package com.google.cloud.genomics.dataflow.utils;
+
+import static com.google.api.services.storage.StorageScopes.DEVSTORAGE_READ_ONLY;
+
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.genomics.GenomicsScopes;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.StorageScopes;
+import com.google.api.services.storage.model.StorageObject;
+import com.google.cloud.genomics.utils.GenomicsFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.security.GeneralSecurityException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.annotation.Nullable;
+
+/**
+ * A helper class to download parts of files.
```

Comment:
From a quick glance there is some overlap in functionality between this class and [DataflowJavaSDK . . . util/gcsio/GoogleCloudStorageReadChannel.java](https://github.com/GoogleCloudPlatform/DataflowJavaSDK/blob/master/sdk/src/main/java/com/google/cloud/dataflow/sdk/util/gcsio/GoogleCloudStorageReadChannel.java) but this class supports simple use cases.

Since the primary use case for this is tests, consider moving it into `src/test` (makes it more clear that for Dataflow pipelines, the Dataflow GCS source should be used)